span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/ExcludeSpanRuleEvaluator.java

```diff
@@ -0,0 +1,181 @@
+package org.hypertrace.core.spannormalizer.jaeger;
+
+import static org.hypertrace.core.span.constants.v1.Http.HTTP_REQUEST_URL;
+import static org.hypertrace.core.span.constants.v1.OTSpanTag.OT_SPAN_TAG_HTTP_URL;
+import static org.hypertrace.core.spannormalizer.jaeger.JaegerSpanNormalizer.OLD_JAEGER_SERVICENAME_KEY;
+
+import io.jaegertracing.api_v2.JaegerSpanInternalModel;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.apache.commons.lang3.StringUtils;
+import org.hypertrace.core.datamodel.AttributeValue;
+import org.hypertrace.core.semantic.convention.constants.http.OTelHttpSemanticConventions;
+import org.hypertrace.core.span.constants.RawSpanConstants;
+import org.hypertrace.core.span.constants.v1.Http;
+import org.hypertrace.core.spannormalizer.util.JaegerHTTagsConverter;
+import org.hypertrace.span.processing.config.service.v1.ExcludeSpanRule;
+import org.hypertrace.span.processing.config.service.v1.Field;
+import org.hypertrace.span.processing.config.service.v1.LogicalOperator;
+import org.hypertrace.span.processing.config.service.v1.LogicalSpanFilterExpression;
+import org.hypertrace.span.processing.config.service.v1.RelationalOperator;
+import org.hypertrace.span.processing.config.service.v1.RelationalSpanFilterExpression;
+import org.hypertrace.span.processing.config.service.v1.SpanFilter;
+
+public class ExcludeSpanRuleEvaluator {
+  // rename and replace SpanFilter later.
+
+  public ExcludeSpanRuleEvaluator() {
+    // empty constructor
+  }
+
+  private static final List<String> FULL_URL_ATTRIBUTES =
+      List.of(
+          RawSpanConstants.getValue(OT_SPAN_TAG_HTTP_URL),
+          RawSpanConstants.getValue(HTTP_REQUEST_URL),
+          RawSpanConstants.getValue(Http.HTTP_URL),
+          OTelHttpSemanticConventions.HTTP_URL.getValue());
+
+  private static final String ENVIRONMENT_ATTRIBUTE = "deployment.environment";
+
+  public boolean shouldDropSpan(
+      Optional<String> tenantIdKey,
+      JaegerSpanInternalModel.Span jaegerSpan,
+      List<ExcludeSpanRule> excludeSpanRules,
```

Comment:
It should not be taking exclude rules as an input. Instead, it should be directly getting the exclude rules from config service, via the caching client